### PR TITLE
remove the nanoS support in boilerplate

### PR DESCRIPTION
**Pull request by tdejoigny-ledger**

### CODE REVIEW

**Comment by agrojean-ledger** [comment]
@tdejoigny-ledger something is wrong with the CI

**Comment by tdejoigny-ledger** [comment]
> @tdejoigny-ledger something is wrong with the CI
I suppose it will be fixed right after the merge.

File: .github/workflows/codeql_checks.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     name: Analyse
     strategy:
       matrix:
-        sdk: [ "$NANOS_SDK", "$NANOX_SDK", "$NANOSP_SDK" ]
+        sdk: ["$NANOX_SDK", "$NANOSP_SDK", "$STAX_SDK", "$FLEX_SDK"]
```
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply .github/workflows/codeql_checks.yml:21]: Also we'll need to do this change in nanos branch too

File: .vscode/c_cpp_properties.json [hunk 2]
```diff
@@ -1,19 +1,19 @@
 {
     "env": {
-        "BOLOS_SDK": "/opt/nanos-secure-sdk",
+        "BOLOS_SDK": "/opt/nanosplus-secure-sdk",
```
Comment (fbeutin-ledger) [inline .vscode/c_cpp_properties.json:3]: What is this file and why is it commited? Suspicious to see only s+ info here

**Review by fbeutin-ledger — CHANGES_REQUESTED** [review]
We'll need to add the new nanos branch on all workflow triggers
```
on:
  push:
    branches:
      - master
      - main
      - develop
      - TODO
      ```
     
We need to do this on nanos branch **AND** on develop branch

**Comment by fbeutin-ledger** [comment]
I suggest we split this PR in two commits:
- C1: first one to fix the tools (codeql, workflow, triggers)
- C2: second one to drop lns support

Then we can re-do the fork lns from the clean commit C1

File: README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 # Ledger Boilerplate Application
 
-This is a boilerplate application which can be forked to start a new project for the Ledger Nano S/X/SP and Stax.
+This is a boilerplate application which can be forked to start a new project for the Ledger Nano X/SP, Stax and Flex.
```
Comment (fbeutin-ledger) [inline README.md:6]: Add a comment about the dedicated nanos branch

File: .vscode/c_cpp_properties.json [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
         "BOLOS_ENV": "/opt/bolos-devenv"
     },
     "configurations": [
```
Comment (fbeutin-ledger) [inline .vscode/c_cpp_properties.json:6]: Why is this list hardcoded?
This should come from the extension?

File: README.md [hunk 5]
```diff
@@ -77,12 +77,11 @@ make DEBUG=1  # compile optionally with PRINTF
 
 You can choose which device to compile and load for by setting the `BOLOS_SDK` environment variable to the following values :
 
-* `BOLOS_SDK=$NANOS_SDK`
 * `BOLOS_SDK=$NANOX_SDK`
 * `BOLOS_SDK=$NANOSP_SDK`
 * `BOLOS_SDK=$STAX_SDK`
 
-By default this variable is set to build/load for Nano S.
+By default this variable is set to build/load for Nano S+.
```
Comment (fbeutin-ledger) [inline README.md:84]: Do we want this behavior? Nothing => no build ?

File: .vscode/c_cpp_properties.json (hunk 4 above)
Comment (tdejoigny-ledger) [reply .vscode/c_cpp_properties.json:6]: Not sure, I suppose that this list contains the default value at Vscode start-up
In any case, cannot be nanos

File: README.md (hunk 3 above)
Comment (tdejoigny-ledger) [reply README.md:6]: nanos branch is only used for internal usage

File: README.md (hunk 5 above)
Comment (tdejoigny-ledger) [reply README.md:84]: I can remove this line

**Review by fbeutin-ledger — APPROVED** [review]

**Pull request merged**